src/Features/ERDDAP/hooks/TableDAPComponents.tsx

```diff
@@ -72,14 +76,48 @@ export const UseDatasets: React.FunctionComponent<UseDatasetsProps> = ({
       }
     }
   }
+  let updatedPlatforms
+
+  if (loadedDatasets && platformId) {
+    const platforms: any = queryClient.getQueryData(["buoybarn-platforms"])
+    if (platforms) {
+      const platform = platforms.features.find((f) => f.id === platformId)
+      const latestReading = loadedDatasets.map((d) => {
+        return {
+          name: d.name,
+          latestValue: d.timeSeries[d.timeSeries.length - 1].reading,
+          time: d.timeSeries[d.timeSeries.length - 1].time.toISOString(),
+        }
+      })
+
+      const updatedReadings = platform.properties.readings.map((reading) => {
+        const update = latestReading.find((r) => r.name === reading.variable)
+        return update ? { ...reading, value: update.latestValue, time: update.time } : { ...reading }
+      })
+
+      const updatedPlatform = {
+        ...platform,
+        properties: {
+          ...platform.properties,
+          readings: updatedReadings,
+        },
+      }
+
+      updatedPlatforms = {
+        ...platforms,
+        features: platforms.features.map((f) => (f.id === platformId ? updatedPlatform : f)),
+      }
+      // queryClient.setQueryData(['buoybarn-platforms'], platforms);
+    }
```

Comment:
How about comparing the reading times, and setting a flag if there is any updated readings or not? 

If there are updated readings, then call `setQueryData` and trigger a re-render. Then during the re-render the times should be the same, so it shouldn't get called again, and thus avoid re-rendering.